dialog/file_test.go

```diff
@@ -282,3 +283,29 @@ func TestFileFilters(t *testing.T) {
 	}
 	assert.Equal(t, 4, count)
 }
+
+func TestFileFavorites(t *testing.T) {
+	win := test.NewWindow(widget.NewLabel("Content"))
+
+	dlg := NewFileOpen(func(reader fyne.URIReadCloser, err error) {
+		assert.Nil(t, err)
+		assert.Nil(t, reader)
+	}, win)
+
+	dlg.Show()
+
+	favorites := dlg.dialog.loadFavorites()
+	places := dlg.dialog.loadPlaces()
+	assert.Equal(t, 3+len(places), len(favorites))
+
+	curDir := ""
+	for _, f := range favorites {
+		btn := f.(*widget.Button)
+		t.Log(fmt.Sprint("Changing from ", curDir, " to ", btn.Text))
+		test.Tap(btn)
+		assert.NotEqual(t, curDir, dlg.dialog.dir)
```

Comment:
Would it not be better to assert that the correct directory is now set?

Reply:
I don't know the exact name of the directory since it depends on the username and language.

Reply:
I would have thought it existed in the map returned by favouriteLocations()?
I understand that it might not be a trivial lookup/comparison - but by not doing it you run the risk that this is simply setting the un-localised path which would be a failure and yet this test would pass...

Reply:
Yes, the paths are inside the map, but using those would be like checking strings against themselves. The unlocalised default paths are used if the xdg-user-dir command doesn't exist or if some other error occurs. The default paths are also used if the xdg-user-dir command returns the home directory. This happens when eg. the documents folder doesn't exist. This is the case for the CI tests.

Reply:
But this test is testing what happens when you tap the button.
To check that the favourites are created correctly is the remit of another test (one that does not create a file dialog UII).
Testing the dialog's new location against the string representing the location that should have been loaded when a button was pressed seems quite correct to me.
If your concern is that the localisation is not tested if we do that then yes another test can be added. But by ignoring the contents of the string I don't see that anything is being added.

Reply:
I changed it so that ```dlg.dialog.dir``` is checked against ```getFavoriteLocations```.

Reply:
> To check that the favourites are created correctly is the remit of another test

What test is this exactly? I couldn't find a test which is testing the favorites.

Reply:
I did not mean that it existed, I was more hinting that tests should test one thing, so maybe a test is missing?

Reply:
I think that this would be wrapped up nicely if we could add a test of the file_xdg contents - do you think that is possible? (clearly they would only run on linux :) )

Reply:
I did my best, but it is difficult to test this since the paths depend on the languange and sometimes don't even exist.